spec/bar-chart-spec.js

```
@@ -949,7 +949,43 @@ describe('dc.barChart', function () {
             });
         });
     });
+    describe('with elasticX and x-axis padding', function() {
+        var date = makeDate(2012, 5, 1);
+        beforeEach(function () {
 
+            var rows = [
+              {x: date, y: 4},
+            ];
+            data = crossfilter(rows);
+            dimension = data.dimension(function (d) {
+              return d.x;
+            });
+            group = dimension.group().reduceSum(function (d) {
+              return d.y;
+            });
+            chart = dc.barChart('#' + id);
+            chart.width(500)
+              .transitionDuration(0)
+              .x(d3.time.scale())
+              .elasticY(true).elasticX(true)
+              .dimension(dimension)
+              .group(group)
+            chart.render();
+        });
+        it ('should render the right xAxisMax/Min when no padding', function () {
+            expect(chart.xAxisMin()).toEqual(date);
+            expect(chart.xAxisMax()).toEqual(date);
+        })
+        it ('should render the right xAxisMax/Min when 10 day padding', function () {
+          chart.xAxisPadding(10)
+          let expectedStartDate = d3.time.day.offset(date, -10);
+          let expectedEndDate = d3.time.day.offset(date, 10);
+          expect(chart.xAxisMin()).toEqual(expectedStartDate);
+          expect(chart.xAxisMax()).toEqual(expectedEndDate);
+        });
+
+        it ('should render the right ')
```

No problem, I understood that. I'll wait for you to comment before I merge. 

Travis went ahead and tested it, and I commented because I was alarmed that our current tools will silently drop tests on the floor. There was another time this happened due to use of an ES6 feature but `let` doesn't appear to be a problem this time. 

I wonder if we need to upgrade our build tools yet again. :-/